### PR TITLE
add more options to update_all_products.pl + fix for products without lc, bug #2214

### PR DESCRIPTION
**Pull request by stephanegigandet**

### CODE REVIEW

File: scripts/update_all_products.pl [hunk 1]
```diff
@@ -20,7 +20,7 @@
 # You should have received a copy of the GNU Affero General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-use Modern::Perl '2012';
+use Modern::Perl '2017';
```
Comment (VaiTon) [inline scripts/update_all_products.pl]: Do we want to mix up Modern 2012 & 2017?

File: scripts/update_all_products.pl [hunk 2]
```diff
@@ -217,6 +245,22 @@
 			}
 		}
 
+		# Fix products that were created without the lc field, but that have a lang field
+		if ($fix_missing_lc) {
```
Comment (hangy) [inline scripts/update_all_products.pl]: Wouldn't it be cleaner to use `if ($fix_missing_lc and (not defined $product_ref->{lc})) {` here and simplify the following `if` statements? There's probably not a huge advantage in printing `$product_ref->{lang}` for every product?

File: scripts/update_all_products.pl [hunk 3]
```diff
@@ -217,6 +245,22 @@
...
+			print STDERR "lang: " . $product_ref->{lang} . "\n";
+			if ((not defined $product_ref->{lc}) and (defined $product_ref->{lang}) and ($product_ref->{lang} =~ /^[a-z][a-z]$/)) {
+				print STDERR "fixing missing lc, using lang: " . $product_ref->{lang} . "\n";
+				$product_ref->{lc} = $product_ref->{lang};
+				$product_values_changed = 1;
+			}
+			elsif (not defined $product_ref->{lc}) {
+				print STDERR "fixing missing lc, lang also missing, assignin en";
```
Comment (hangy) [inline scripts/update_all_products.pl]:
```suggestion
				print STDERR "fixing missing lc, lang also missing, assigning en\n";
```
typo

File: scripts/update_all_products.pl (hunk 1 above)
Comment (VaiTon) [reply scripts/update_all_products.pl]: #2217 has been merged so no longer a problem.

File: scripts/update_all_products.pl (hunk 2 above)
Comment (stephanegigandet) [reply scripts/update_all_products.pl]: Right, I will change it, thanks!

**Pull request merged**